src/main/java/com/puppycrawl/tools/checkstyle/checks/header/HeaderCheck.java

```diff
@@ -88,7 +90,8 @@ public void setIgnoreLines(int... list) {
     }
 
     @Override
-    protected void processFiltered(File file, List<String> lines) {
+    protected void processFiltered(File file, FileText fileText) {
+        final List<String> lines = fileText.getLines();
```

Comment:
please avoid usage of temporal object - List<String> lines , that is easy refactoring.